test/regressions/dg-solvency-simulation.t.sol

```diff
@@ -0,0 +1,1656 @@
+
+            uint256 shareRateAfter = _lido.stETH.getPooledEthByShares(10 ** 18);
+            uint256 balanceEstimated = holderBalanceBefore * (shareRateAfter + accidentalLockedErr) / shareRateBefore;
+
+            assert(holderBalanceAfter >= holderBalanceBefore);
+            assert(holderBalanceAfter <= balanceEstimated);
+        }
+    }
+
+    function _calculateLockedSharesAndUnstETHInEscrow(
+        Escrow escrow,
+        address account
+    ) internal view returns (uint256 sharesLocked, uint256 unstETHUnclaimed) {
+        Escrow.VetoerDetails memory vetoerDetails = escrow.getVetoerDetails(account);
+        sharesLocked = vetoerDetails.stETHLockedShares.toUint256();
+
+        uint256[] memory unstETHIds = escrow.getVetoerUnstETHIds(account);
+
+        IWithdrawalQueue.WithdrawalRequestStatus[] memory withdrawalStatuses =
+            _lido.withdrawalQueue.getWithdrawalStatus(unstETHIds);
+        Escrow.LockedUnstETHDetails[] memory unstETHDetails = escrow.getLockedUnstETHDetails(unstETHIds);
+
+        for (uint256 i = 0; i < unstETHDetails.length; ++i) {
+            if (unstETHDetails[i].status != UnstETHRecordStatus.Withdrawn) {
+                unstETHUnclaimed += withdrawalStatuses[i].amountOfStETH;
+            }
+        }
+    }
+
+    function _getRandomSimulationAccount() internal returns (address) {
+        return _simulationAccounts[_random.nextUint256(SIMULATION_ACCOUNTS)];
+    }
+
+    function _getSimulationAccount(uint256 index) internal returns (address) {
+        string memory accountName = string(bytes.concat("SIMULATION_ACC_", bytes(Strings.toString(index))));
+        return makeAddr(accountName);
+    }
+
+    function _submitStETHByRandomAccount(address[] storage accounts) internal {
+        // console.log(">>> Submitting stETH by random account");
+        uint256 randomIndexOffset = _random.nextUint256(accounts.length);
+
+        for (uint256 i = 0; i < accounts.length; ++i) {
+            address account = accounts[(randomIndexOffset + i) % accounts.length];
+
+            uint256 balance = account.balance;
+
+            if (balance < MIN_ST_ETH_SUBMIT_AMOUNT) {
+                return;
+            }
+
+            uint256 submitAmount =
+                _random.nextUint256(MIN_ST_ETH_SUBMIT_AMOUNT, Math.min(balance, MAX_ST_ETH_SUBMIT_AMOUNT));
+
+            vm.prank(account);
+            _lido.stETH.submit{value: submitAmount}(address(0));
+
+            _totalSubmittedStETH += submitAmount;
+
+            _accountsDetails[account].stETHSubmitted += submitAmount;
+
+            // console.log("Account %s submitted %s stETH.", account, submitAmount.formatEther(), balance.formatEther());
+        }
+    }
+
+    function _submitWstETHByRandomAccount(address[] storage accounts) internal {
+        // console.log(">>> Submitting wstETH by random account");
+        uint256 randomIndexOffset = _random.nextUint256(accounts.length);
+
+        for (uint256 i = 0; i < accounts.length; ++i) {
```

Comment:
Same : "Here happens submission by all simulation accounts, but it is expected to be only one random account"